src/sock/socket_session.hpp

```diff
@@ -40,10 +41,26 @@ class outbound_message
     virtual std::string_view buffer() = 0;
 };
 
+/*
+* Use this to keep in track of different thresholds which we need to deal with. e.g - maximum amount of bytes allowed per minute through a session
+* threshold_limit - Maximum threshold value which is allowed
+* counter_value - Counter which keeps incrementing per every message
+* timestamp - Timestamp when counter value changes
+* interval_in_ms - Time interval in miliseconds in which the threshold and the counter value should be compared
+*/
+struct session_thresholds
```

Comment:
The plurarl struct name seems weird. Specially this struct is used to track a single threshold type.